### DISCUSSION TITLE
Radxa Rock Support Status

### DISCUSSION THREAD

**Discussion by THX723** [marcone/teslausb#778] [Q&A]
So what's the latest on TeslaUSB support? Is it ready for prime time?

**Comment by KrzysztofHajdamowicz** [comment]
In my opinion- no, not yet. I still haven't found base image stable that will work with OTG/gadget mode.
Check #776

**Reply by THX723** [reply]
@KrzysztofHajdamowicz Thanks. I appreciate the update. Also curious... are you using Rock 4 or Rock 5 -- not that it has anything to do w OTG?

**Reply by KrzysztofHajdamowicz** [reply]
Rock Pi 4C plus revision 1.4

**Comment by miles267** [comment]
I've been able to successfully setup a Rock Pi 4C+ with connectivity to the top USB3 port.  Unfortunately however it disconnects intermittently to the USB3 and the wifi connectivity (throughput) even with a 2.4/5.0 ghz antenna is poor at best.  Have reverted back to RPi4.

**Reply by THX723** [reply]
Ouch! Not the update I was looking for.

**Reply by KrzysztofHajdamowicz** [reply]
> wifi connectivity (throughput) even with a 2.4/5.0 ghz antenna is poor at best.

Have you tried Molex antena with u.fl connector? I had huge signal boost when switched from stock (deaf) antenna.

https://allegro.pl/oferta/molex-146153-0100-antena-wifi-2-4-5-ghz-z-u-fl-13999022528

**Reply by miles267** [reply]
Yes. I’m using that exactly antenna. Also it seems to only connect via 5 GHz and never attempts to switch to 2.4 GHz if the signal isn’t strong enough. It’s all bizarre.

**Reply by marcone** [reply]
Current Armbian builds have otg on the top USB3 port by default, but a nonfunctioning bottom USB3 port (it functions as neither otg nor host). This works for TeslaUSB, but if you wanted to use an external drive for storage, you'd have to connect it to one of the USB2 ports.
Unfortunately it looks like Armbian for the Rock Pi 4C+ no longer has an official maintainer, which is probably why this regressed.

**Reply by THX723** [reply]
@marcone That's a real shame.
